src/python/pants/backend/go/pkg.py

```diff
@@ -70,6 +81,68 @@ class ResolvedGoPackage:
     # _test.go files outside package. Obtained from `XTestGoFiles` key in package metadata.
     xtest_go_files: Tuple[str, ...]
 
+    @classmethod
+    def from_metadata(
+        cls,
+        metadata: dict,
+        *,
+        import_path: Optional[str] = None,
+        address: Optional[Address] = None,
+        module_address: Optional[Address] = None,
```

Comment:
What's the difference?

Reply:
`address` is the Address of the `go_package` (if any) that was analyzed, and `None` if produced from a third-party package (although it will get set later once I have hooked up analysis of _go_ext_mod_packages)

`module_address` is the Address of the owning `go_module`

Reply:
Hm, ack that naming is hard: wondering if that can be made more explicit. Something like `owning_go_module: Address` and `go_package: Address`. The type hint makes the `_address` bit redundant.

Reply:
I'm going to just add a TODO for now re the naming. I need to see how the fields in `ResolvedGoPackage` end up as I implement third-party package support.